modules/reactor/src/meshgenerators/HexagonMeshTrimmer.C

```diff
@@ -0,0 +1,566 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "HexagonMeshTrimmer.h"
+#include "MooseMeshUtils.h"
+
+// C++ includes
+#include <cmath> // provides round, not std::round (see http://www.cplusplus.com/reference/cmath/round/)
+
+registerMooseObject("ReactorApp", HexagonMeshTrimmer);
+
+InputParameters
+HexagonMeshTrimmer::validParams()
+{
+  InputParameters params = PolygonMeshGeneratorBase::validParams();
+  params.addRequiredParam<MeshGeneratorName>("input", "The input mesh that needs to be trimmed.");
+  params.addRangeCheckedParam<std::vector<unsigned short>>(
+      "trim_peripheral_region",
+      std::vector<unsigned short>(6, 0),
```

Comment:
should be bools

Reply:
It is easier to check a short vector than a bool vector for error messages. Also vector<bool> is not a supported input parameter time.